components/tidb_query_executors/src/simple_aggr_executor.rs

```diff
@@ -203,7 +215,9 @@ impl<Src: BatchExecutor> AggregationExecutorImpl<Src> for SimpleAggregationImpl
         mut iteratee: impl FnMut(&mut Entities<Src>, &[Box<dyn AggrFunctionState>]) -> Result<()>,
     ) -> Result<Vec<LazyBatchColumn>> {
         assert!(src_is_drained);
-        iteratee(entities, &self.states)?;
+        if self.has_input_rows {
+            iteratee(entities, &self.states)?;
+        }
```

Comment:
Maybe you also need to change L207

Reply:
This [comment](https://github.com/tikv/tikv/blob/a7e89a85bb5661f6ea211b319ea6253341a53197/components/tidb_query_executors/src/util/aggr_executor.rs#L65) indicates that `group_len` can be inaccurate since it is just a hint. But I've updated L207 as you suggested.